src/library/trackcollection.cpp

```diff
@@ -510,4 +516,68 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    struct stat sb; //this struct will hold the access info of the dir using struct here as below it has function of same name
+    stat(m_pConfig->getSettingsPath().toStdString().c_str(), &sb);  //give the status, m_pConfig->getSettingsPath() will give the
```

Comment:
Third answerer: 
str.toLocal8Bit().constData()

That is correct for Linux and Mac. 
If such a construct is required on windows as well, whatch out for a wchar version for windows Posix API cases exists.
```
int _wstat(  
   const wchar_t *path,  
   struct _stat *buffer   
);  
```